src/Build/BackEnd/Components/Scheduler/Scheduler.cs

```diff
@@ -2280,8 +2271,37 @@ private void AssignGlobalRequestId(BuildRequest request)
                 }
             }
 
-            request.GlobalRequestId = _nextGlobalRequestId;
-            _nextGlobalRequestId++;
+            // No matching request found, assign a new ID
+            request.GlobalRequestId = _nextGlobalRequestId++;
+        }
+
+        /// <summary>
+        /// Computes a hash code for a collection of targets, ignoring order and case.
+        /// </summary>
+        private int ComputeTargetsHash(List<string> targets)
```

Comment:
I'm curious how much this buys us compared to just calling `SetEquals`

Reply:
For clarification, the build in string hash code functions are somewhat expensive since they enumerate the whole string. Having this check requires enumerating each string in the list to accumulate the comparison hash code, and we then do essentially the same work inside of `SetEquals`.

Reply:
makes a lot of sense, applied the suggestion